### PR TITLE
Issue 187 and 207

### PR DESCRIPTION
**Pull request by vsoftco**
Fixed

### CODE REVIEW

File: .travis/global-namespace-check.sh [hunk 1]
```diff
@@ -1,11 +1,11 @@
 #!/bin/bash
 
-if [[ $(nm -g liboqs.a | grep ' T ' | grep -E -v -i ' T [_]?[OQS|ntru|picnic|Keccak]') ]];
+if [[ $(nm -g liboqs.a | grep ' T ' | grep -E -v -i ' T [_]?(OQS|ntru|picnic|Keccak|.*SIKEp503|.*SIDHp503|.*SIKEp751|.*SIDHp751|.*shake128|.*shake256|rand_bytes|print_hex|cpu_supports|uint64_from_char_array|uint64_to_char_array|_ntt_double|_rec|_aligned_alloc|_aligned_free)') ]];
 then
 	tput setaf 1;
 	echo "Code contains the following non-namespaced global symbols; see https://github.com/open-quantum-safe/liboqs/wiki/Coding-conventions for function naming conventions.";
 	tput sgr 0
-	nm -g liboqs.a | grep ' T ' | grep -E -v -i ' T [_]?[OQS|ntru|picnic|Keccak]'
+  nm -g liboqs.a | grep ' T ' | grep -E -v -i ' T [_]?(OQS|ntru|picnic|Keccak|.*SIKEp503|.*SIDHp503|.*SIKEp751|.*SIDHp751|.*shake128|.*shake256|rand_bytes|print_hex|cpu_supports|uint64_from_char_array|uint64_to_char_array|_ntt_double|_rec)'
```
Comment (dstebila) [inline .travis/global-namespace-check.sh]: I'd be worried about a few of these being popular names, including print_hex, _ntt_double, and _rec.  What's the motivation for including these?

Also, is there a reason that line 8's regex is slightly different than line 3's?  Perhaps we should make the regex a variable and then use that variable on both lines.

Comment (vsoftco) [reply .travis/global-namespace-check.sh]: I refactored the regex into the shell variable REGEX. Not sure what was the initial motivation of disabling `print_hex`, `_ntt_double`, and `_rec`, so I kept them. @smashra any idea?

**Comment by smashra** [comment]
I'm not sure why they were not there.
But when I updated the regex I added them.

**Comment by dstebila** [comment]
But why did you add them?  They are the types of functions we probably ought to be catching and renaming.

**Comment by smashra** [comment]
I think I got them mixed up with the third party contributions which are
supposed to be added to the regex.
I'll update.

**Comment by smashra** [comment]
print_hex symbol is coming from  Christian's code:

====================================
 ./src/sig_picnic/external/io.c:void print_hex(FILE* out, const uint8_t*
data, size_t len) {
./src/sig_picnic/external/picnic_impl.c:  print_hex(out, msg, msglen);
./src/sig_picnic/external/picnic_impl.c:  print_hex(out, sig, siglen);
./src/sig_picnic/external/picnic_impl.c:  print_hex(out, sig,
challenge_size);
./src/sig_picnic/external/picnic_impl.c:    print_hex(out,
round->commitments[2], digest_size);
./src/sig_picnic/external/picnic_impl.c:      print_hex(out, round->gs[2],
unruh_g_len);
./src/sig_picnic/external/picnic_impl.c:    print_hex(out,
round->communicated_bits[1], view_size);
./src/sig_picnic/external/picnic_impl.c:    print_hex(out, round->seeds[0],
seed_size);
./src/sig_picnic/external/picnic_impl.c:    print_hex(out, round->seeds[1],
seed_size);
./src/sig_picnic/external/picnic_impl.c:      print_hex(out,
round->input_shares[1], input_size);
./src/sig_picnic/external/picnic_impl.c:      print_hex(out,
round->input_shares[0], input_size);
./src/sig_picnic/external/tools/visualize_signature.c:  print_hex(stdout,
&sk.data[1], blocksize);
./src/sig_picnic/external/tools/visualize_signature.c:  print_hex(stdout,
&pk.data[1], 2 * blocksize);
./src/sig_picnic/external/tools/visualize_signature.c:  print_hex(stdout,
&pk.data[1], blocksize);
./src/sig_picnic/external/tools/visualize_signature.c:  print_hex(stdout,
&pk.data[1 + blocksize], blocksize);
============================================

I'll check in the updated global namespace check file.

**Comment by vsoftco** [comment]
Ready to merge. I kept 'print_hex' for now, as it's part of Picnic. The regex is now refactored into the shell variable $REGEX.

**Comment by vsoftco** [comment]
@dstebila The PR is ready to merge, all tests passed

**Pull request merged**